File: built-in plugins dsp/mb-chorus-vintage.cpp
/**
 * MB Vintage Chorus
 * Category : effect
 * Type     : chorus
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : 80s vintage chorus
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_CHORUS_VINTAGE_H
#define MB_CHORUS_VINTAGE_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbChorusVintage : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-chorus-vintage";
    static constexpr const char* PLUGIN_NAME    = "MB Vintage Chorus";
    static constexpr const char* PLUGIN_TYPE    = "chorus";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float rate = 0.8f;  // range [0.1, 5]
    float depth = 0.6f;  // range [0, 1]
    float mix = 0.5f;  // range [0, 1]
    };

    MbChorusVintage() = default;
    ~MbChorusVintage() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.rate = std::clamp(params.rate, 0.1f, 5f);
        params.depth = std::clamp(params.depth, 0f, 1f);
        params.mix = std::clamp(params.mix, 0f, 1f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Vintage Chorus
        return input;
    }
};

#endif // MB_CHORUS_VINTAGE_H

File: built-in plugins dsp/mb-chorus-vintage.ts
import type { PluginDefinition } from "../server/services/pluginHostService";

const MbChorusVintagePlugin: PluginDefinition = {
  id: "mb-chorus-vintage",
  slug: "mb-chorus-vintage",
  name: "MB Vintage Chorus",
  category: "effect",
  type: "chorus",
  version: "1.0.0",
  description: "80s vintage chorus",
  author: "Max Booster",
  grade: "A",
  parameters: [
    {
      id: "rate",
      name: "Rate",
      type: "float",
      defaultValue: 0.8,
      minValue: 0.1,
      maxValue: 5,
      automatable: true,
    },
    {
      id: "depth",
      name: "Depth",
      type: "float",
      defaultValue: 0.6,
      minValue: 0,
      maxValue: 1,
      automatable: true,
    },
    {
      id: "mix",
      name: "Mix",
      type: "float",
      defaultValue: 0.5,
      minValue: 0,
      maxValue: 1,
      automatable: true,
    },
  ],
  defaultPreset: { rate: 0.8, depth: 0.6, mix: 0.5 },
};

export default MbChorusVintagePlugin;
